common/appointment/cancel.ts

```diff
@@ -17,32 +19,40 @@ export async function cancelAppointment(user: User, appointment: Appointment) {
 
     const student = await getStudent(user);
 
-    switch (appointment.appointmentType) {
-        case AppointmentType.group:
-            const subcourse = await prisma.subcourse.findFirst({ where: { id: appointment.subcourseId }, include: { course: true } });
-            const participants = await prisma.subcourse_participants_pupil.findMany({ where: { subcourseId: subcourse.id }, include: { pupil: true } });
-
-            for (const participant of participants) {
-                await Notification.actionTaken(participant.pupil, 'student_cancel_appointment_group', {
+    // Notifications are sent only when an appointment is cancelled, not when a subcourse is cancelled
+    if (!silent) {
+        switch (appointment.appointmentType) {
+            case AppointmentType.group:
+                const subcourse = await prisma.subcourse.findFirst({ where: { id: appointment.subcourseId }, include: { course: true } });
+                const participants = await prisma.subcourse_participants_pupil.findMany({ where: { subcourseId: subcourse.id }, include: { pupil: true } });
+
+                for (const participant of participants) {
+                    await Notification.actionTaken(participant.pupil, 'student_cancel_appointment_group', {
+                        appointment: getAppointmentForNotification(appointment),
+                        student,
+                        course: subcourse.course,
+                    });
+                }
+
+                break;
+
+            case AppointmentType.match:
+                const match = await prisma.match.findUnique({ where: { id: appointment.matchId }, include: { pupil: true } });
+                await Notification.actionTaken(match.pupil, 'student_cancel_appointment_match', {
                     appointment: getAppointmentForNotification(appointment),
                     student,
-                    course: subcourse.course,
                 });
-            }
-
-            break;
 
-        case AppointmentType.match:
-            const match = await prisma.match.findUnique({ where: { id: appointment.matchId }, include: { pupil: true } });
-            await Notification.actionTaken(match.pupil, 'student_cancel_appointment_match', {
-                appointment: getAppointmentForNotification(appointment),
-                student,
-            });
+                break;
 
-            break;
-
-        case AppointmentType.internal:
-        case AppointmentType.legacy:
-            break;
+            case AppointmentType.internal:
+            case AppointmentType.legacy:
+                break;
+        }
+    }
+    if (appointment.zoomMeetingId) {
+        await deleteZoomMeeting(appointment);
+    } else {
+        throw new RedundantError('Appointment has no Zoom Meeting');
```

Comment:
Is this an error condition? If we cancel the appointment without a zoom meeting, then just cancelling it is fine?